### ISSUE TITLE
Ensure PDS_SCAN_TARGET_URL never ends with a slash

### ISSUE THREAD

**Issue by de-jcup** [mercedes-benz/sechub#899]
### Situation

URI defined within a webScan configuration are already given to `PDS `as `PDS_SCAN_TARGET_URL ` :

https://github.com/Daimler/sechub/blob/cbd90fede4a1894bfc05cbec1bd278bc5c0c6356/sechub-adapter-pds/src/main/java/com/daimler/sechub/adapter/pds/PDSWebScanConfigImpl.java#L85

So when somebody defineds `https://example.com/myApp1` it does not end with a slash but, defining `https://example.com/myApp1/` it does end.
Inside #568 we want to provide includes without the base URI, so easier to read. But PDS implementations should not need have logic inside to determine if the baseURI ends with a slash or not.
So with #568  we will need to ensure the given PDS URI does not end with a slash - so PDS scripts can rely on the data given from Secub ENV entry.

### Wanted
 `PDS_SCAN_TARGET_URL `  contains always a URL string which does ALWAYS not end with slash

### Solution
Change `PDSWebScanConfigImpl` accordingly and implement some tests

**Comment by Vivek-Prajapatii** [comment]
Interested in it

**Comment by Jeeppler** [comment]
@Vivek-Prajapatii go for it. In case, you have any questions feel free to ask.

**Comment by Vivek-Prajapatii** [comment]
ok i understood! 
So i need to append a code which will remove the slash(/) from the end of the URI, if present.
did i understood in the right way, if not kindly ping me.

**Comment by Jeeppler** [comment]
@Vivek-Prajapatii yes, that is exactly right. 

A URL which looks like this: `https://example.org/` -> will become -> `https://example.org`. 
Please write the code and a test. We use Junit4 and Junit5 in our code base. Please use [Junit5](https://junit.org/junit5/) for new tests.

There are currently 2500+ tests. Most of them are unit tests (testing the functionality of a class or method). However, a few are integration tests (testing parts or the entire system). In your case writing a unit test is enough.

Please consider valid and non valid test cases. For example, the following should work:

1. `https://example.org/` -> `https://example.org`
2. `https://example.org/#` -> `https://example.org`
3. `http://example.org/` -> `http://example.org`
4. `http://example.org/#` -> `http://example.org`
5. `https://example.org/?search=abc` ->  `https://example.org`
…

@winzj  and @de-jcup do you have some additional examples? What about `https://example.org/index.html` or `https://example.org/customer/profile/` etc.?

**Comment by Vivek-Prajapatii** [comment]
> @Vivek-Prajapatii yes, that is exactly right.
> 
> A URL which looks like this: `https://example.org/` -> will become -> `https://example.org`. Please write the code and a test. We use Junit4 and Junit5 in our code base. Please use [Junit5](https://junit.org/junit5/) for new tests.
> 
> There are currently 2500+ tests. Most of them are unit tests (testing the functionality of a class or method). However, a few are integration tests (testing parts or the entire system). In your case writing a unit test is enough.
> 
> Please consider valid and non valid test cases. For example, the following should work:
> 
> 1. `https://example.org/` -> `https://example.org`
> 2. `https://example.org/#` -> `https://example.org`
> 3. `http://example.org/` -> `http://example.org`
> 4. `http://example.org/#` -> `http://example.org`
> 5. `https://example.org/?search=abc` ->  `https://example.org`
>    …
> 
> @winzj and @de-jcup do you have some additional examples? What about `https://example.org/index.html` or `https://example.org/customer/profile/` etc.?

i have written the code but i dont know hoe to write the tests in junit

**Comment by Vivek-Prajapatii** [comment]
done with it

**Comment by Jeeppler** [comment]
@Vivek-Prajapatii thank you. I linked your Pull-Request to this issue.

**Comment by de-jcup** [comment]
The origin problem why I created this issue was:

> While we implemented our OWASP Zap java wrapper for SecHub, the written ZAP wrapper did use the `PDS_SCAN_TARGET_URL` directly without any checks. But OWASP ZAP does not like URLs ending with `/` - so we ensure URLs do not end with /

But after thinking about the problem while doing the PR review the following came up : The problem is absolutely product specific! Maybe we got another tool in future where we want the origin URL or we need with slashes at the end etc. etc.
Also `PDS_SCAN_TARGET_URL` is not webscan specific, but could be useful for infrascan as well - where we could need more information etc. So it should be handled by the product - because only the product does know which corner cases should be handled.

So long story short: We will just keep as is and the OWASP ZAP handling of URL ending will be done by our Java based ZAP wrapper on product site (PDS).

@Vivek-Prajapatii : Thanx for the PR and your effort - I will you give feedback there, even when we do not need the feature anymore

**Comment by Vivek-Prajapatii** [comment]
ok i understand
